### PR TITLE
Add test to make sure policy dataclass configs match yaml configs

### PR DESCRIPTION
**Pull request by alexander-soare**
## What this does

^

### CODE REVIEW

File: lerobot/common/policies/factory.py [hunk 1]
```diff
@@ -75,7 +81,9 @@ def make_policy(
             policy. Therefore, this argument is mutually exclusive with `pretrained_policy_name_or_path`.
     """
     if not (pretrained_policy_name_or_path is None) ^ (dataset_stats is None):
-        raise ValueError("Only one of `pretrained_policy_name_or_path` and `dataset_stats` may be provided.")
+        raise ValueError(
+            "Exactly one of `pretrained_policy_name_or_path` and `dataset_stats` must be provided."
```
Comment (alexander-soare) [inline lerobot/common/policies/factory.py:85]: FYI, changed wording from "only" to "exactly" to be true to the logic

File: lerobot/common/policies/factory.py [hunk 2]
```diff
@@ -86,9 +94,10 @@ def make_policy(
     else:
         # Load a pretrained policy and override the config if needed (for example, if there are inference-time
         # hyperparameters that we want to vary).
-        # TODO(alexander-soare): This hack makes use of huggingface_hub's tooling to load the policy with, pretrained
-        # weights which are then loaded into a fresh policy with the desired config. This PR in huggingface_hub should
-        # make it possible to avoid the hack: https://github.com/huggingface/huggingface_hub/pull/2274.
+        # TODO(alexander-soare): This hack makes use of huggingface_hub's tooling to load the policy with,
```
Comment (alexander-soare) [inline lerobot/common/policies/factory.py:97]: FYI: just formatting to keep within line length.

File: lerobot/configs/policy/tdmpc.yaml [hunk 3]
```diff
@@ -54,7 +54,7 @@ policy:
   discount: 0.9
 
   # Inference.
-  use_mpc: false
+  use_mpc: true
```
Comment (alexander-soare) [inline lerobot/configs/policy/tdmpc.yaml:57]: FYI: technically this is b/c breaking, but I don't think anyone's using TD-MPC yet as it hasn't been announced to be fully functional

File: tests/scripts/save_policy_to_safetensors.py [hunk 4]
```diff
@@ -108,13 +108,18 @@ def save_policy_to_safetensors(output_dir, env_name, policy_name, extra_override
 
 if __name__ == "__main__":
     env_policies = [
-        # ("xarm", "tdmpc", []),
+        # ("xarm", "tdmpc", [], ""),
         # (
         #     "pusht",
         #     "diffusion",
-        #     ["policy.n_action_steps=8", "policy.num_inference_steps=10", "policy.down_dims=[128, 256, 512]"],
+        #     [
```
Comment (alexander-soare) [inline tests/scripts/save_policy_to_safetensors.py:115]: FYI, adding the `file_name_extra` everywhere.

**Review by Cadene — APPROVED** [review]
Thanks! LGTM

**Pull request merged**